Missing `make_temporary_platform_primitives` declaration when building on M1 Macs

**Problem**: After a fresh checkout, installation and `cargo build` running `cargo test` fails on M1 Macs with:

```
error[E0432]: unresolved import `crate::platform::make_temporary_platform_primitives`
   --> rust/noosphere/src/sphere/builder.rs:361:26
    |
361 |         key::KeyStorage, platform::make_temporary_platform_primitives, sphere::SphereContext,
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `make_temporary_platform_primitives` in `platform`
```

**Potential Solution**

This is due to a missing declaration for this function on the M1 platform which should be added in platform.rs: https://github.com/subconsciousnetwork/noosphere/blob/96a3752c1f70e8d088293c98d2638b67f1a733f5/rust/noosphere/src/platform.rs#L10-L21

**Context**

```
➜ rustc -vV
rustc 1.67.1 (d5a82bbd2 2023-02-07)
binary: rustc
commit-hash: d5a82bbd26e1ad8b7401f6a718a9c57c96905483
commit-date: 2023-02-07
host: aarch64-apple-darwin
release: 1.67.1
LLVM version: 15.0.6
```

macOS 13.2, M1 Macbook Pro

--- a/rust/noosphere/src/platform.rs
+++ b/rust/noosphere/src/platform.rs
@@ -18,6 +18,27 @@ mod inner {
     pub type PlatformKeyMaterial = Ed25519KeyMaterial;
     pub type PlatformKeyStorage = InsecureKeyStorage;
     pub type PlatformStorage = NativeStorage;
+
+    #[cfg(test)]
+    use anyhow::Result;
+
+    #[cfg(test)]
+    use std::path::PathBuf;
+
+    #[cfg(test)]
+    use tempfile::TempDir;
+
+    #[cfg(test)]
+    pub async fn make_temporary_platform_primitives(
+    ) -> Result<(PathBuf, PlatformKeyStorage, (TempDir, TempDir))> {
+        let sphere_dir = TempDir::new().unwrap();
+
+        let key_dir = TempDir::new().unwrap();
+
+        let key_storage = InsecureKeyStorage::new(key_dir.path())?;
+
+        Ok((sphere_dir.path().into(), key_storage, (sphere_dir, key_dir)))
+    }
 }
 
 #[cfg(target_arch = "wasm32")]

